_posts/2021-12-13-hhast-4.139.0.markdown

```diff
@@ -0,0 +1,108 @@
+---
+title: "How to lint your code with the help of the type checker?"
+layout: post
+author: atry
+category: blog
+---
+
+HHAST is a toolkit for for analyzing the structure of Hack files. It includes a linting and migration framework to find and automatically fix problems in your code. The lint errors and migration suggestions can be accessed from Hack API, command line interface or IDEs.
+
+HHAST also ships with many built-in lint rules and migration rules. However, previously the HHAST built-in lint rules have limited access to the type checker, therefore, it is difficult to create a lint rule that needs to determine questions like "is this type a subtype of that type?" or "is this expression compatible with that function signature?".
+
+Recently, we released HHAST v4.139.0. The new version brings a proxy to run the linter from the Hack compiler written in OCaml, which includes more powerful lint rules that need the access to the type checker [^1].
+
+## How to enable lint rules from the Hack compiler?
```

Comment:
Maybe: How to enable typechecker-based lint rules

It doesn't involve the compiler

Reply:
Section title changed